### PR TITLE
Remove all hard-coded macros from examples

### PR DESCRIPTION
**Pull request by trombik**

### CODE REVIEW

**Comment by trombik** [comment]
@UncleRus i would like to modify all the I2C examples (adding [Kconfig.projbuild](https://github.com/UncleRus/esp-idf-lib/pull/299/files#diff-966a0799e6e8ca68e57b414c5eab9ebb02c582d07c8ee023904873b16d3b0021)) so that users do not have to modify hard-coded macros. what do you think?

**Comment by nated0g** [comment]
Good idea @trombik .

I've updated the SCD30 library example, following your updates to the other libraries exactly.

Should I make a separate PR for that?

**Comment by trombik** [comment]
@nated0g thank you. the branch is open only for maintainers to commit. if you have a patch, paste your patch somewhere such as gist.github.com. i'll merge it. otherwise, there would be duplicated efforts. i'll work on I2C-based drivers, `hd44780`, `ina219`, `ina3221`, and `lm75`.

**Comment by nated0g** [comment]
Branch with the changes I made to the scd30 lib are here:

https://github.com/nated0g/esp-idf-lib/tree/scd30-example-kconfig

File: examples/hd44780_i2c/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+In most cases, the `HD44780` module comes with a `PCF8574` module designed for
+`HD44780`. The `PCF8574` module has pins for I2C and `hd44780`. In that case,
+you simply connect two modules. The example assumes a typical configuration,
+described below.
+
+| `hd44780` | `PCF8574` |
+|-----------|-----------|
+| `RS`      | 0         |
+| `E`       | 1         |
```
Comment (trombik) [inline examples/hd44780_i2c/README.md:22]: i do not have a PCF8574 module at hand, but this should be `2`. users who use bare PCF8574 can easily modify wiring, but those who use a PCF8574 module cannot.

Comment (UncleRus) [reply examples/hd44780_i2c/README.md:22]: It is 2:
```C
    hd44780_t lcd = {
        .write_cb = write_lcd_data, // use callback to send data to LCD by I2C GPIO expander
        .font = HD44780_FONT_5X8,
        .lines = 2,
        .pins = {
            .rs = 0,
            .e  = 2,
            .d4 = 4,
            .d5 = 5,
            .d6 = 6,
            .d7 = 7,
            .bl = 3
        }
    };
```

Comment (trombik) [reply examples/hd44780_i2c/README.md]: you are right. not enough coffee that day.

**Comment by trombik** [comment]
i'm working on `lm75`, `ina3221`, and `max31725`.

File: examples/ina219/main/Kconfig.projbuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            Expected max current in Ampare. Note that the value must be int, not float.
+
+    config EXAMPLE_SHUNT_RESISTOR_MILLI_OHM
+        int "Shunt resistor value in milliohm)"
+        default 100
+        help
+            Shunt resistor value in milliohm. Must not be zero.
+
+    config EXAMPLE_I2C_ADDR
+        hex "I2C address of PCF8574"
```
Comment (UncleRus) [inline examples/ina219/main/Kconfig.projbuild:15]: INA219

File: examples/ina219/main/main.c [hunk 3]
```diff
@@ -31,7 +26,8 @@ void task(void *pvParameters)
             INA219_RES_12BIT_1S, INA219_RES_12BIT_1S, INA219_MODE_CONT_SHUNT_BUS));
 
     ESP_LOGI(TAG, "Calibrating INA219");
-    ESP_ERROR_CHECK(ina219_calibrate(&dev, 5.0, 0.1)); // 5A max current, 0.1 Ohm shunt resistance
+
+    ESP_ERROR_CHECK(ina219_calibrate(&dev, (float)CONFIG_EXAMPLE_MAX_CURRENT, (1 / (float)CONFIG_EXAMPLE_SHUNT_RESISTOR_MILLI_OHM)));
```
Comment (UncleRus) [inline examples/ina219/main/main.c:30]: Should be
```C
ESP_ERROR_CHECK(ina219_calibrate(&dev, (float)CONFIG_EXAMPLE_MAX_CURRENT, (float)CONFIG_EXAMPLE_SHUNT_RESISTOR_MILLI_OHM / 1000.0f));
```

**Comment by UncleRus** [comment]
Examples:

- [x] ads1115
- [x] aht
- [x] bh1750
- [ ] bh1900nux
- [x] bme680
- [x] bmp180
- [x] bmp280
- [x] button
- [x] ccs811
- [x] dht
- [x] ds1302
- [x] ds1307
- [ ] ds18x20_multi
- [ ] ds18x20_single
- [x] ds3231
- [x] ds3502
- [ ] encoder
- [ ] hd44780_gpio
- [x] hd44780_i2c
- [x] hd44780_i2c_scroll
- [ ] hdc1000
- [x] hmc5883l
- [ ] hts221
- [x] hx711
- [x] ina219
- [ ] ina260
- [x] ina3221
- [ ] led_effects
- [x] led_strip
- [x] led_strip_spi
- [x] lm75
- [ ] ls7366r
- [ ] max31725
- [ ] max31865
- [ ] max7219_7seg
- [ ] max7219_8x8
- [ ] mcp23017
- [x] mcp23s17
- [ ] mcp342x
- [ ] mcp4725
- [ ] mcp960x
- [ ] mcp9808
- [ ] mhz19b
- [x] ms5611_i2c
- [ ] pca9557
- [ ] pca9685
- [ ] pcf8563
- [ ] pcf8575
- [ ] pcf8591
- [x] qmc5883l
- [ ] rda5807m
- [x] scd30
- [ ] scd4x
- [ ] sgp40
- [ ] sht3x
- [ ] sht4x
- [x] si7021
- [ ] simplest_barometer
- [ ] tca95x5
- [ ] tda74xx
- [ ] tsl2561
- [ ] tsl2591_interrupt
- [ ] tsl2591_simple
- [ ] tsl4531
- [ ] tsys01
- [x] ultrasonic
- [x] wiegand_reader

**Comment by trombik** [comment]
sorry for the recent inactivity in the PR. I'm still willing to work on the PR, but my time is bit limited right now.

**Comment by UncleRus** [comment]
> sorry for the recent inactivity in the PR. I'm still willing to work on the PR, but my time is bit limited right now.

It's alright, no need to apologize.

**Comment by UncleRus** [comment]
I'm currently working on `tca96x5`

**Review by trombik — COMMENTED** [review]
this commit was deliberately NOT `git rebase`d as that would have remove all the test history, and i am not the only person who is working on the branch.

**Review by trombik — COMMENTED** [review]
I am trying to read and understand every example code, but i am not sure sometimes that my understanding is correct. i appreciate reviews like this.

**Comment by UncleRus** [comment]
I'm going to merge this PR. The unfinished can be completed later.

I want to say a BIG THANK YOU to @trombik!

**Comment by trombik** [comment]
fine with me. I'll make another PR later.

**Pull request merged**